spec/models/authenticator/ldap_spec.rb

```diff
@@ -155,6 +156,23 @@ def normalize(dn)
     end
   end
 
+  describe '#authorize_queue' do
+    def authenticate
+      subject.authenticate(username, password)
+    end
+
+    let(:username) { 'alice' }
+    let(:password) { 'secret' }
+    let(:config) { super().merge(:ldap_role => true) }
+
+    it "encrypts password for queuing" do
+      allow(MiqTask).to receive(:create).and_return(double(:id => :return_value))
+      allow(MiqQueue).to receive(:put) { :return_value }
+      authenticate
+      expect(config[:bind_pwd]).to match(/v([0-9]+):\{([^}]*)\}/)
+    end
+  end
```

Comment:
This spec is testing a private method; IMO you should move this to a nested context under `#authenticate` below. 

```ruby
describe "#authenticate" do
  context "when using LDAP" do
     ...
  end
```

As for the the encryption, testing the format of the encryption is out of scope of the spec. You should just ensure that encryption is called for properly. I'd just `allow(MiqPassword).to_receive(:try_encrypt).and_return("LOLZ ENCRYPTED!")` and `expect(config[:bind_pwd]).to eq("LOLZ ENCRYPTED!")`